src/pages/VenuePage/TemplateWrapper.tsx

```diff
@@ -90,6 +93,11 @@ const TemplateWrapper: React.FC<TemplateWrapperProps> = ({ venue }) => {
 
   return (
     <WithNavigationBar fullscreen={fullscreen}>
+      <UserProfileModal
+        userProfile={selectedUserProfile}
+        show={selectedUserProfile !== undefined}
+        onHide={() => setUserProfile(undefined)}
+      />
       <AnnouncementMessage message={venue?.bannerMessage} />
       {template}
       <ChatSidebar />
```

Comment:
[nit] I tend to put modals at the bottom of things, both to 'keep them out of the way' and because 'last items' in render order tend to appear on top of earlier items (not that that would make a difference here)

```suggestion
      <AnnouncementMessage message={venue?.bannerMessage} />
      {template}
      <ChatSidebar />
      <UserProfileModal
        userProfile={selectedUserProfile}
        show={selectedUserProfile !== undefined}
        onHide={() => setUserProfile(undefined)}
      />
```